spec/acceptance/suites/tar_el7/01_simp_server_spec.rb

```diff
@@ -0,0 +1,174 @@
+      end
+
+      it 'should provide default hieradata to make beaker happy' do
+        create_remote_file(master, '/etc/puppetlabs/code/environments/simp/hieradata/default.yaml', {
+          'sudo::user_specifications' => {
+            'vagrant_all' => {
+              'user_list' =>  ['vagrant'],
+              'cmnd'      =>  ['ALL'],
+              'passwd'    =>  false,
+            },
+          },
+          'pam::access::users' => {
+            'defaults' => {
+              'origins'    => ['ALL'],
+              'permission' =>  '+'
+            },
+            'vagrant' => nil
+          },
+          'ssh::server::conf::permitrootlogin'    =>  true,
+          'ssh::server::conf::authorizedkeysfile' =>  '.ssh/authorized_keys',
+          # The following settings are because $server_facts['serverip'] is
+          # incorrect in a beaker/vagrant (mutli-interface) environment
+          'simp::puppet_server_hosts_entry'       => false,
+          'simp::rsync_stunnel'                   => master_fqdn,
+          # Make sure puppet doesn't run (hopefully)
+          'pupmod::agent::cron::minute'           => '0',
+          'pupmod::agent::cron::hour'             => '0',
+          'pupmod::agent::cron::weekday'          => '0',
+          'pupmod::agent::cron::month'            => '1',
+          }.to_yaml
+        )
+      end
+      it 'should enable autosign' do
+        on(master, 'puppet config --section master set autosign true')
+      end
+
+      it 'should run simp bootstrap' do
+        # Remove the lock file because we've already added the vagrant user stuff
+        on(master, 'rm -f /root/.simp/simp_bootstrap_start_lock')
+
+        on(master, 'simp bootstrap --no-verbose -u --remove_ssldir > /dev/null')
+      end
+
+      it 'should reboot the host' do
+        master.reboot
+        sleep(240)
+      end
+      it 'should have puppet runs with no changes' do
+        on(master, '/opt/puppetlabs/bin/puppet agent -t', :acceptable_exit_codes => [0,2,4,6])
+        on(master, '/opt/puppetlabs/bin/puppet agent -t', :acceptable_exit_codes => [0] )
+      end
+      it 'should generate agent certs' do
+        togen = []
+        agents.each do |agent|
+          togen << agent.hostname + '.' + domain
+        end
+        create_remote_file(master, '/var/simp/environments/production/FakeCA/togen', togen.join("\n"))
+        on(master, 'cd /var/simp/environments/production/FakeCA; ./gencerts_nopass.sh')
+      end
+    end
+  end
+
+  # context 'classify nodes' do
+  # end
+
+  context 'agents' do
+    agents.each do |agent|
+      it 'should install the agent' do
+        use_puppet_repo = ENV['BEAKER_puppet_repo'] || false
+        if use_puppet_repo
+          if agent.host_hash[:platform] =~ /el-7/
+            agent.install_package('http://yum.puppetlabs.com/puppetlabs-release-pc1-el-7.noarch.rpm')
+          else
+            agent.install_package('http://yum.puppetlabs.com/puppetlabs-release-pc1-el-6.noarch.rpm')
```

Comment:
These should not be hard coded. Puppet 5 vs Puppet 4 is completely different. May need to update `simp-beaker-helpers` to handle this.